src/java.base/share/classes/sun/security/util/DerValue.java

```diff
@@ -956,13 +1145,39 @@ public void resetTag(byte tag) {
         this.tag = tag;
     }
 
+    /**
+     * Returns a new DerValuw with a different tag.
+     *
+     * @param newTag the new tag
+     * @return a new DerValue
+     */
+    public DerValue withTag(byte newTag) {
```

Comment:
Seems like a somewhat dangerous method. The value may not match the new tag? The caller is expected to know what it's doing?

Reply:
I intend to use it to turn back IMPLICIT element to its original tag, otherwise getXyz() calls would fail because they are checking the tag. In the old design, checking the tag (in DerValue) and reading the value (in DerInputBuffer) are separated but now they are in a single method (in DerValue).

The caller pattern should always look like `withTag(INTEGER).getInteger()`.

Reply:
Hmm, I see, maybe add more comments then.